external-crates/move/move-compiler/src/diagnostics/codes.rs

```diff
@@ -14,13 +14,26 @@ pub enum Severity {
     Bug = 3,
 }
 
+/// Identifies a warning category. Includes an option external prefix to distinguish warnings from
+/// different sources.
+#[derive(PartialEq, Eq, Clone, Copy, Debug, Hash, PartialOrd, Ord)]
+pub struct CategoryID {
+    category: u8,
+    external_prefix: Option<&'static str>,
```

Comment:
Nit,  maybe a type alias for this, then we wouldn't need some of the comments below (maybe)